distributed/diagnostics/progress.py

```diff
@@ -124,7 +124,9 @@ def stop(self, exception=None, key=None):
             self.scheduler.plugins.remove(self)
         if exception:
             self.status = "error"
-            self.extra.update({"exception": self.scheduler.exceptions[key], "key": key})
+            self.extra.update(
+                {"exception": self.scheduler.tasks[key].exception, "key": key}
+            )
```

Comment:
Grepping the code, I also found this invocation on scheduler plugin side which still used the legacy state mapping. This is not a functional change but merely avoids the StateLegacyMapping on scheduler side.